desc/configuration.py

```diff
@@ -1023,12 +1067,43 @@ def iota(self, new):
     @property
     def i_l(self):
         """Coefficients of iota profile (ndarray)."""
-        return self.iota.params
+        return np.empty(0) if self.iota is None else self.iota.params
 
     @i_l.setter
     def i_l(self, i_l):
+        if self.iota is None:
+            raise ValueError(
+                "Attempt to set rotational transform on an equilibrium with fixed toroidal current"
+            )
         self.iota.params = i_l
 
+    @property
+    def current(self):
+        """Toroidal current (I) profile."""
+        return self._current
+
+    @current.setter
+    def current(self, new):
```

Comment:
Couldn't hurt to raise an `AttributeError` if we attempt setting `current` when `current` is already `None`. This would imply the equilibrium has been created with some fixed `iota` and something is trying to fix `current` to some value.

Likewise for setting `iota`.

Reply:
Not sure, I could see someone wanting to switch an equilibrium from fixed current to fixed iota. Speaking of, we should probably add a method to get either profile (computing the values of necessary)

Reply:
Yea I thought of that in my edit. But do we have a specified precedence on which profile to fix if both become not None? I'd choose the most recently set one.

Reply:
I think we could have a method to return either profile, and allow an equilibrium to have both assigned, but throw an error when solving if both are fixed. Remember that the actual fixing is done by the `FixIota` or `FixCurrent` constraints. So we'd probably want to throw the error if both of those are included in a solve.

Reply:
It's true that the constraint determines which mode is used during the equilibrium solve, either `FixIota` or `FixCurrent`. So the user could assign both profiles to an Equilibrium and then choose which to use by specifying the constraint. But the present code will default to using iota as the constraint if both profiles are assigned.

Reply:
I think one of the last things we need is some sort of getter method for the current/iota profile that would allow for converting between them. Any thoughts on API?

``get_current(self, grid=None, profile_type="spline")``
``get_iota(self, grid=None, profile_type="spline")``

or something more generic?

``get_profile(self, name, grid=None, profile_type="spline")``

where ``name`` could be current, iota, pressure, or maybe others as well?